internal/types/types_test.go

```diff
@@ -0,0 +1,58 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHumanizeDuration(t *testing.T) {
+	testCases := []struct {
+		name     string
+		input    int
+		expected string
+	}{
+		{
+			name:     "0 seconds",
+			input:    0,
+			expected: "0s",
+		},
+		{
+			name:     "30 seconds",
+			input:    30,
+			expected: "30s",
+		},
+		{
+			name:     "60 seconds",
+			input:    60,
+			expected: "1m",
+		},
+		{
+			name:     "1805 seconds",
+			input:    1805,
+			expected: "30m",
+		},
+		{
+			name:     "3605 seconds",
+			input:    3605,
+			expected: "1h",
+		},
+		{
+			name:     "4200 seconds",
+			input:    4200,
+			expected: "1h 10m",
+		},
+		{
+			name:     "87000 seconds",
+			input:    87000,
+			expected: "24h 10m",
```

Comment:
It's great you added this tests file

It's great as you apparently prepare a refactoring based on what I suggested

Maybe you should:
- consider removing the minutes when duration is bigger than x days
  I mean having 27h 13m might be meaningless

So here you could apply

```go
duration = duration.Truncate(1 * time.Hour)
```

- also, you might consider reporting days after a number of hours (36, 48 maybe)

Then you can use int(duration.Hours/24) to get the number of "days", and a modulo 24 for the remaining hours

Reply:
Given the use case (people tracking time on a task for a while), the amount tracked should not go above several hours at a time (in most cases). As such, I want "hour" to be the largest unit of time I use to show a time period. So, I guess this function suffices that use case, for now.